### PR TITLE
CLEWS-15676: add option to index by series in get_df, use it in GDH 

### PR DESCRIPTION
**Pull request by nemozen**
this index gives users a shortcut to retrieve one series at a time from within the df

Screenshot from 2020-03-30 14-31-31

Screenshot from 2020-03-30 14-31-58

Screenshot from 2020-03-30 14-32-16

### CODE REVIEW

File: api/client/gro_client.py [hunk 1]
```diff
@@ -257,19 +259,16 @@ def GDH(self, gdh_selection, **optional_selections):
...
         # add optional pararms to selection
         for key, value in list(optional_selections.items()):
-            if key not in entity_keys:
+            if key not in DATA_SERIES_UNIQUE_COLS:
                 selection[key] = value
 
         self.add_single_data_series(selection)
-        df = self.get_df()
-        return df
+        return self.get_df(index_by_series=True).loc[[tuple(entity_ids)], :]
```
Comment (muzigao) [inline api/client/gro_client.py]: this `.loc[[tuple(entity_ids)], :]` part will throw an error if given selection doesn't have at-level data. e.g client.GDH("860032-10009-1215-0-14-9")

File: api/client/gro_client.py [hunk 2]
```diff
@@ -64,13 +63,18 @@ def get_df(self, show_revisions=False):
...
-
+            If index_by_series is set, the dataframe is indexed by series.
+            See https://developers.gro-intelligence.com/data-series-definition.html
         """
         while self._data_series_queue:
             data_series = self._data_series_queue.pop()
             if show_revisions:
                 data_series['show_revisions'] = True
             self.add_points_to_df(None, data_series, self.get_data_points(**data_series))
+        if index_by_series:
```
Comment (muzigao) [inline api/client/gro_client.py:74]: tested with a couple of examples, data frame multi-index  part looks good to me !

File: api/client/gro_client.py (hunk 1 above)
Comment (nemozen) [reply api/client/gro_client.py]: that's the intended behavior... gdh is expected to return only that series exactly.

Comment (muzigao) [reply api/client/gro_client.py]: yeah I understand that - but it's showing 

`KeyError: "None of [MultiIndex([(860032, 10009, 1215, 0, 14, 9)],\n           names=['metric_id', 'item_id', 'region_id', 'partner_region_id', 'source_id', 'frequency_id'])] are in the [index]"`

should we return an empty data frame or something ?

**Review by stephenbernardi — APPROVED** [review]
LGTM. In the future, can we add an improvement to convert index ids to names?

File: api/client/gro_client.py (hunk 1 above)
Comment (nemozen) [reply api/client/gro_client.py]: done

**Review by muzigao — APPROVED** [review]

**Pull request merged**